homeassistant/components/modbus/const.py

```diff
@@ -66,6 +68,7 @@
 CONF_VERIFY = "verify"
 CONF_VERIFY_REGISTER = "verify_register"
 CONF_VERIFY_STATE = "verify_state"
+CONF_ZERO_CLAMP_VALUE = "zero_clamp_value"
```

Comment:
I believe zero suppress stops zero (suppresses it) which is the opposite to what this is doing.
I have seen the description zero_elevation_threshold, i.e. it elevates zero evaluation to a higher level
In engineering and computers I am more familiar with clamping e.g. https://learn.microsoft.com/en-us/dotnet/api/system.math.clamp?view=net-7.0

Reply:
No it is exactly what you are doing, you suppress detection of small values, by replacing them by a real zero.

Quoting Microsoft which is an information company, seems a bit out of scope. Companies like Siemens and ABB are industry leaders in modbus, they are the ones you should quote. I should know since I have been in the world of SCADA systems for a lifetime with these companies.

We try to use modbus terminology to configure the modbus integration, because when people read the manual for their modbus device, they find the same names and thus making it easier for them.

Reply:
I was trying to find anything in Siemens and I think they call it deadband?
https://support.industry.siemens.com/forum/WW/en/posts/deadband-function-in-pid-controllers/94125
https://www.manualslib.com/manual/1654917/Siemens-Sinamics-G120xa.html?page=145
How about deadband_value ?

Reply:
Deadband is a term used primarily for transmission technology, not very often for measuring technology. Most important a deadband is not fixed to be around zero, but e.g. for transmissions are based on the gauss curve.

Why invent new names, when we have a name that covers the functionality ? I believe strongly that having a name like "zero_suppresion" tells the user what to expect.

Reply:
I want to get this right, use industry standard names, and learn .... do you have any reference to zero suppression in Siemens?

I have been looking for links and could only find deadzone for flattening noise around zero.

I was just considering a nontechnical audience where suppressing something is to prevent it, so suppressing zero will prevent zero from happening ... the naming seems wrong .. 100% opposite to what we want - however I agree that sometimes industry standard names are counter-intuitive !

Reply:
I too, as codeowner, wants to get this right, but I am not going into a war of who can make the best quotes, I honestly have more important things to do.

Funny enough your considering is exactly mine….you suppress the detection of small values around zero.

Discussing further seems like loosing time (which I do not have). The PR is ready to be approved apart from the name and the missing documentation, the rest only depend on you.